### PR TITLE
Removed the decorator

### PR DESCRIPTION
**Pull request by dantleech**
This PR removes the `dantleech/decorator-factory` and uses the new "provider" feature of the MassiveSearchBundle instead.

Why? Basically the MassiveSearchBundle, like alot of other bundles, requires metadata that is mapped to classes. With the current system the class is always `PageDocument` for example, and the structure type cannot be inferred from this information.

It is therefore necessary to either create a unique class for each Document/StructureType combination, or to rethink the metadata mapping in the SearchBundle.

The current approach is to (in a very hacky way) generate a unique class for each combination. This could be done in a cleaner way at the DocumentManager level, and it could be that that is the best way, but it requires more thought.

But for now we can ignore we can ignore this and see if it really is a requirement later.

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php [hunk 1]
```diff
@@ -189,6 +188,12 @@ public function loadMetadataForClass(\ReflectionClass $class)
             ));
         }
 
+        $indexMeta->addFieldMapping(self::FIELD_STRUCTURE_TYPE, array(
+            'type' => 'string',
+            'index_strategy' => Field::INDEX_STORED_INDEXED,
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php]: This should be STORED, NOT INDEXED. Wait for https://github.com/massiveart/MassiveSearchBundle/pull/44 to be merged.

Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php]: @dantleech The other PR is fixed, so this could be fixed now, right?

**Comment by danrot** [comment]
To me it doesn't really make sense to review that, since nobody has reviewed the PR you want to merge this into anyway until now.

**Comment by dantleech** [comment]
Merging this. When we do upgrade the massive search bundle this will break automatically and will need to be fixed.

**Pull request merged**